module/Company/src/Company/Service/Company.php

```diff
@@ -2,52 +2,454 @@
 
 namespace Company\Service;
 
-use Application\Service\AbstractService;
+//use Application\Service\AbstractService;
+use Application\Service\AbstractAclService;
 
-use Company\Model\Company as CompanyModel;
-use Company\Mapper\Company as CompanyMapper;
 /**
  * Company service.
  */
-class Company extends AbstractService
+class Company extends AbstractACLService
 {
+    /**
+     * Returns an list of all companies (excluding hidden companies
+     *
+     */
     public function getCompanyList()
     {
-        return $this->getCompanyMapper()->findAll();
+        $translator = $this->getTranslator();
+        if ($this->isAllowed('list')) {
+            return $this->getCompanyMapper()->findAll($translator->getLocale());
+        } else {
```

Comment:
It is better to avoid using `else`. You can simply invert the if condition, and then throw the exception.

About the `else` thing, read this: http://williamdurand.fr/2013/06/03/object-calisthenics/#2-don-t-use-the-else-keyword (maybe everyone should actually read this entire document once, @jszanto @mstruijs @wsinnema @stefanmajoor @gafderks ).